### DISCUSSION TITLE
LoadAnnotations code  [Copy frome Wechat]

### DISCUSSION THREAD

**Discussion by vansin** [open-mmlab/mmdetection#8900] [Q&A]
@五喵-菜狗闻星 麻烦你看一下 3.x中 LoadAnnotations类中_load_bboxes函数中的results['gt_ignore_flags'] = np.array(gt_ignore_flags, dtype=np.bool) 会变成bool类型但 PackDetInputs类中的transform函数中“”if 'gt_ignore_flags' in results:
    valid_idx = np.where(results['gt_ignore_flags'] == 0)[0]
    ignore_idx = np.where(results['gt_ignore_flags'] == 1)[0]“”这段代码会有问题吧

![image](https://user-images.githubusercontent.com/25839884/192733107-521071a5-fc83-4362-9dc6-b1754fe1c17f.png)

**Comment by zgq91** [comment]
在LoadAnnotations类 _load_bboxes函数中 “results['gt_ignore_flags'] = np.array(gt_ignore_flags, dtype=np.bool)” 将results['gt_ignore_flags']数据类型转换为bool；但在PackDetInputs类 transform函数中 
            valid_idx = np.where(results['gt_ignore_flags'] == 0)[0]
            ignore_idx = np.where(results['gt_ignore_flags'] == 1)[0]  改判断会导致 valid_idx和ignore_idx都为空
